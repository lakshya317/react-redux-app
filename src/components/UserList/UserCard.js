import React from 'react';

const UserCard = (props) => {
    const {user} = props;
    return (
        <div className="user-card-container">
            <img src={user.avatar} alt={user.first_name} className="user-card-avatar"/>
            <div className='user-info-container'>
                <div className="user-card-name">
                    {`${user.first_name} ${user.last_name}`}
                </div>
                <div className="user-card-email">
                    {user.email}
                </div>
            </div>
        </div>
    );
}

export default UserCard;
