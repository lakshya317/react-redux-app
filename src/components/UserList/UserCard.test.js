import { render } from '@testing-library/react';
import UserCard from './UserCard';
import { MemoryRouter } from 'react-router-dom';

describe('User Card Tests', () => {
    test('User Card Snapshot Test', () => {
        const userCard = render(
            <MemoryRouter>
                <UserCard
                    user={{
                        id: 1,
                        email: 'george.bluth@reqres.in',
                        first_name: 'George',
                        last_name: 'Bluth',
                        avatar: 'https://reqres.in/img/faces/1-image.jpg'
                    }}
                />
            </MemoryRouter>
        );

        expect(userCard).toMatchSnapshot();
    });

    test('Name and Email test verification', () => {
        const userCard = render(
            <MemoryRouter>
                <UserCard
                    user={{
                        id: 1,
                        email: 'george.bluth@reqres.in',
                        first_name: 'George',
                        last_name: 'Bluth',
                        avatar: 'https://reqres.in/img/faces/1-image.jpg'
                    }}
                />
            </MemoryRouter>
        );

        userCard.getByText('George Bluth');
        userCard.getByText('george.bluth@reqres.in');
    });
});
