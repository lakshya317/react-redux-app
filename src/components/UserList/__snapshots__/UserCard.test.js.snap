// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Card Tests User Card Snapshot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="user-card-container"
      >
        <img
          alt="George"
          class="user-card-avatar"
          src="https://reqres.in/img/faces/1-image.jpg"
        />
        <div
          class="user-info-container"
        >
          <div
            class="user-card-name"
          >
            George Bluth
          </div>
          <div
            class="user-card-email"
          >
            george.bluth@reqres.in
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="user-card-container"
    >
      <img
        alt="George"
        class="user-card-avatar"
        src="https://reqres.in/img/faces/1-image.jpg"
      />
      <div
        class="user-info-container"
      >
        <div
          class="user-card-name"
        >
          George Bluth
        </div>
        <div
          class="user-card-email"
        >
          george.bluth@reqres.in
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
